src/lava/utils/float2fixed.py

```diff
@@ -1,26 +1,719 @@
+                    except (TypeError, ValueError):
+                        raise ValueError(f'Error in {t_port} of Proc {p_id}')
+                    t_port_impl_shift = t_port_conv_data['implicit_shift']
+                    # If there has been not implicit shift set so far set one.
+                    if not p_impl_shift:
+                        p_impl_shift = t_port_impl_shift
+
+                    # Check for consistency of implicit shifts due to Ports
+                    elif not p_impl_shift == t_port_impl_shift:
+                        raise ValueError(
+                            "Implicit shift due to connections inconsistent in"
+                            + f" Process {p_id}. Cannot perform conversion.")
+
+            for var in p_var_ports['Var']:
+                # Get LavaPyType instance of var.
+                var_py_type = p_model.__getattribute__(var)
+                # Skip meta parameters since they don't need to be converted.
+                if var_py_type.meta_parameter:
+                    continue
+
+                # Retrieve conversion data.
+                try:
+                    conv_data[p_id][var] = var_py_type.conversion_data()
+                except (TypeError, ValueError):
+                    raise ValueError(f'Error in {var} of Proc {p_id}')
+
+                # Update implicit shift if Variable is on global scale domain.
+                if conv_data[p_id][var]['scale_domain'] == 0:
+                    conv_data[p_id][var]['implicit_shift'] += p_impl_shift
+
+                # If Variable var is constant and domain is None store initial
+                # value in domain.
+                update_domain = (
+                    conv_data[p_id][var]['constant']
+                    and conv_data[p_id][var]['domain'] is None
+                )
+                if update_domain:
+                    init = self.procs[p_id].__getattribute__(var).init
+                    conv_data[p_id][var]['domain'] = init
+
+                # Keep track of which scale domain has variables in Processes.
+                self.scale_domains.add(
+                    (conv_data[p_id][var]['scale_domain'], p_id))
+
+        return conv_data
+
+    def _create_monitoring_infrastructure(self) -> dict:
+        """Create and connect monitors for determining the dynamic range of
+        variables in the model that needs to be converted.
+        Only variables which are neither a meta parameter nor a constant nor
+        have a predefined dynamic range will be recorded.
+        Monitors are stored in a nested dictionary with key Process Id, key
+        Variable name.
+
+        Returns
+        -------
+        monitors : dict
+            Monitoring infrastructure for Float2Fixed converter in nested dict
+            {ProcID:
+                {VarName:
+                     monitor}}
+        """
+        monitors = {}
+
+        for p_id, conv_data_p_id in self.conv_data.items():
```

Comment:
Here, it should be possible to replace nested for-loops by a dict comprehension.

Reply:
Same point as above - readability is I think more important.  
But I am open to your opinion whether or not to change it.